clang/test/lab1/veselov_mikhail/test.cpp

```diff
@@ -0,0 +1,212 @@
+// NON_EXIST_VAR-NEXT: int b = 3;
+// NON_EXIST_VAR-NEXT: b += a;
+// NON_EXIST_VAR-NEXT: a++;
+// NON_EXIST_VAR-NEXT:  return b - a;
+// NON_EXIST_VAR-NEXT: }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/VeselRenamePlugin%pluginext\
+// RUN: -add-plugin rename\
+// RUN: -plugin-arg-rename type=func\
+// RUN: -plugin-arg-rename cur-name=function\
+// RUN: -plugin-arg-rename new-name=new_func %t/rename_func.cpp
+// RUN: FileCheck %s < %t/rename_func.cpp --check-prefix=FUNC
+
+// FUNC: int new_func(int param) {
+// FUNC-NEXT: int a;
+// FUNC-NEXT: a = 2;
+// FUNC-NEXT: return param + a;
+// FUNC-NEXT: }
+// FUNC-NEXT: int other_func(){
+// FUNC-NEXT: new_func(3);
+// FUNC-NEXT: int a = new_func(2) + 3;
+// FUNC-NEXT: return a;
+// FUNC-NEXT: }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/VeselRenamePlugin%pluginext\
+// RUN: -add-plugin rename\
+// RUN: -plugin-arg-rename type=func\
+// RUN: -plugin-arg-rename cur-name=function\
+// RUN: -plugin-arg-rename new-name=f %t/rename_non_existent_func.cpp
+// RUN: FileCheck %s < %t/rename_non_existent_func.cpp --check-prefix=NON_EXIST_FUNC
+
+// NON_EXIST_FUNC: int func(int a) {
+// NON_EXIST_FUNC-NEXT: int b = 2;
+// NON_EXIST_FUNC-NEXT: return a + b;
+// NON_EXIST_FUNC-NEXT: }
+// NON_EXIST_FUNC: void func2() {
+// NON_EXIST_FUNC-NEXT: int c = func(2);
+// NON_EXIST_FUNC-NEXT: int b = func(c) + func(3);
+// NON_EXIST_FUNC-NEXT: }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/VeselRenamePlugin%pluginext\
+// RUN: -add-plugin rename\
+// RUN: -plugin-arg-rename type=class\
+// RUN: -plugin-arg-rename cur-name=Base\
+// RUN: -plugin-arg-rename new-name=SimpleClass %t/rename_class.cpp
+// RUN: FileCheck %s < %t/rename_class.cpp --check-prefix=CLASS
+
+// CLASS: class SimpleClass{
+// CLASS-NEXT: private:
+// CLASS-NEXT: int a;
+// CLASS-NEXT: int b;
+// CLASS-NEXT: public:
+// CLASS-NEXT: SimpleClass() {}
+// CLASS-NEXT: SimpleClass(int a, int b): a(a), b(b) {}
+// CLASS-NEXT: ~SimpleClass();
+// CLASS-NEXT: };
+// CLASS: void func() {
+// CLASS-NEXT: SimpleClass a;
+// CLASS-NEXT: SimpleClass* var = new SimpleClass(1, 2);
+// CLASS-NEXT: delete var;
+// CLASS-NEXT: }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/VeselRenamePlugin%pluginext\
+// RUN: -add-plugin rename\
+// RUN: -plugin-arg-rename type=class\
+// RUN: -plugin-arg-rename cur-name=B\
+// RUN: -plugin-arg-rename new-name=C %t/rename_non_existent_class.cpp
+// RUN: FileCheck %s < %t/rename_non_existent_class.cpp --check-prefix=NON_EXIST_CLASS
+
+// NON_EXIST_CLASS: class A{
+// NON_EXIST_CLASS-NEXT: private:
+// NON_EXIST_CLASS-NEXT: int var1;
+// NON_EXIST_CLASS-NEXT: double var2;
+// NON_EXIST_CLASS-NEXT: public:
+// NON_EXIST_CLASS-NEXT: A() {};
+// NON_EXIST_CLASS-NEXT: ~A() {};
+// NON_EXIST_CLASS-NEXT: };
+// NON_EXIST_CLASS: void func() {
+// NON_EXIST_CLASS-NEXT: A var1;
+// NON_EXIST_CLASS-NEXT: A* var2 = new A;
+// NON_EXIST_CLASS-NEXT: delete var2;
+// NON_EXIST_CLASS-NEXT: }
+
+// RUN: %clang_cc1 -load %llvmshlibdir/VeselRenamePlugin%pluginext\
```

Comment:
Could you please split tests below as well?

Reply:
Sorry, previous comment was improper.
Clarifying: I'd like to see CHECK strings closer to code that they validate.